checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,353 @@
+   */
+  protected static class CheckStatusMapper extends Mapper<Object, Text, Text, Text> {
+    /**
+     * Gets the Status and IP address of each mapper task node.
+     */
+    protected void map(Object key, Text value, Context context)
+        throws IOException, InterruptedException {
+      context.write(new Text(performIntegrationChecks().toString()), new Text(getAddress()));
+    }
+  }
+
+  /**
+   * In each reducer node, we will combine the IP addresses that have the same Status.
+   */
+  protected static class MergeStatusReducer extends Reducer<Text, Text, Text, Text> {
+    private Text mMergedAddress = new Text();
+
+    /**
+     * Merges the IP addresses of same Status.
+     */
+    protected void reduce(Text key, Iterable<Text> values, Context context)
+        throws IOException, InterruptedException {
+      String mergedAddress = "";
+      for (Text val : values) {
+        String nodeAddress = val.toString();
+        if (!mergedAddress.contains(nodeAddress)) {
+          mergedAddress = mergedAddress + " " + nodeAddress;
+        }
+      }
+      mMergedAddress.set(mergedAddress);
+      context.write(key, mMergedAddress);
+    }
+  }
+
+  /**
+   * @return if this current task node can recognize Alluxio classes and filesystem
+   */
+  private static Status performIntegrationChecks() {
+    // Checks if MapReduce nodes can recognize Alluxio classes
+    try {
+      // Checks if MapReduce task nodes can recognize Alluxio common classes
+      Class.forName("alluxio.AlluxioURI");
+      // Checks if MapReduce task nodes can recognize Alluxio core client classes
+      Class.forName("alluxio.client.file.BaseFileSystem");
+      Class.forName("alluxio.hadoop.AlluxioFileSystem");
+    } catch (ClassNotFoundException e) {
+      LOG.error("Failed to find Alluxio classes on classpath ", e);
+      return Status.FAIL_TO_FIND_CLASS;
+    }
+
+    // Checks if MapReduce task nodes can recognize Alluxio filesystem
+    try {
+      FileSystem.getFileSystemClass("alluxio", new Configuration());
+    } catch (Exception e) {
+      LOG.error("Failed to find Alluxio filesystem ", e);
+      return Status.FAIL_TO_FIND_FS;
+    }
+
+    return Status.SUCCESS;
+  }
+
+  /**
+   * @return the current task node IP address
+   */
+  private static String getAddress() {
+    String address;
+    try {
+      address = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      LOG.debug("MapReduce cannot get host address of current MapReduce node.");
+      address = "unknown address";
+    }
+    return address;
+  }
+
+  /**
+   * Creates the Hdfs filesystem to generate input and output files.
+   *
+   * @param conf Hadoop configuration
+   */
+  private static void createHdfsFilesystem(Configuration conf) throws Exception {
+    // Inits HDFS File System Object
+    String systemUserName = System.getProperty("user.name");
+    System.setProperty("HADOOP_USER_NAME", systemUserName);
```

Comment:
I get the system username and create the corresponding path in hdfs (like local user is alluxio, create /user/alluxio/ in hdfs. EMR user is hadoop, create /user/hadoop/ in hdfs) to avoid permission denied problem, but I am not sure if it works in other Hadoop cluster like CDH and HDP.
Another option is to let users define their hdfs username and pass in as an argument.